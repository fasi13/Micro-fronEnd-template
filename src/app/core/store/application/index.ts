export * from './application.effects';
export * from './application.actions';