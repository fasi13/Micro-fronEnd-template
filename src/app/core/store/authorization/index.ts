export * from './authorization.effects';
export * from './authorization.actions';