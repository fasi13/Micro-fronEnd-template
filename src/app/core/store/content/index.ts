export * from './content.effects';
export * from './content.actions';
