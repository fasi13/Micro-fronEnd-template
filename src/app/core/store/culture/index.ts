export * from './culture.actions';
export * from './culture.effects';
