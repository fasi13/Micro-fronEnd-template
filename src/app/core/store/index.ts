export * from './authorization';
export * from './router';
export * from './application';
export * from './store.reducers';