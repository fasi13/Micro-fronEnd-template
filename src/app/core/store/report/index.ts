export * from './report.actions';
export * from './report.effects';
