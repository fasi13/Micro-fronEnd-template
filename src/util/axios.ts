import axios, { AxiosError, AxiosRequestConfig, AxiosResponse } from 'axios';

const HierarchyClient = axios.create({
	baseURL: process.env.REACT_APP_CONTENT_API,
	headers: { 'Content-Type': 'application/json' },
});
export const axiosSuccessRequestInterceptor = (request: AxiosRequestConfig) =>
	request;
export const axiosErrorRequestInterceptor = (error: any) =>
	Promise.reject(error);

export const axiosErrorResponseInterceptor = (error: AxiosError) =>
	Promise.reject(error);
export const axiosSuccessResponseInterceptor = (response: AxiosResponse) =>
	response;

HierarchyClient.interceptors.response.use(
	axiosSuccessResponseInterceptor,
	axiosErrorResponseInterceptor,
);
HierarchyClient.interceptors.request.use(
	axiosSuccessRequestInterceptor,
	axiosErrorRequestInterceptor,
);

export { HierarchyClient };
