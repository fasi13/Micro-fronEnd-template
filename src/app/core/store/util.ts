import _mapValues from 'lodash/mapValues';
import _keyBy from 'lodash/keyBy';
import _lowerCase from 'lodash/lowerCase';

import { State } from "./store.reducers";
import { User, UserToken, Link, MappedLinks } from "../models";
import { ApplicationState } from './application/application.reducers';
import { ContentState } from './content/content.reducers';
import { UserManagementStade } from './user-management/user.management.reducers';

const typeCache: { [label: string]: boolean } = {};
export function ActionType<T>(label: T | string): T {
  if (typeCache[<string>label]) {
    throw new Error(`Action type ${label} is already registered`);
  }

  typeCache[<string>label] = true;

  return <T>label;
}

export function mapLinks(_links: Link[]): MappedLinks {
  return _mapValues(_keyBy(_links, 'rel'), ({ href, method } : Link) => (
    {
      href,
      method: _lowerCase(method.method)
    }
  ));
}

export function loadFromLocalStorage() {
  let user: User = JSON.parse(localStorage.getItem('user'));
  const token: UserToken = localStorage.getItem('token');
  if (user) {
    user = { ...user, actions: mapLinks(user._links) };
  }
  const currentState: State = {
    router: null,
    authorization: {
      authenticated: !!token && !!user,
      loaded: false,
      loading: false,
      creatingUser: false,
      createdUser: false,
      user
    },
    application: {} as ApplicationState,
    content: {} as ContentState,
    userManagement: {} as UserManagementStade
  };
  return currentState;
}
