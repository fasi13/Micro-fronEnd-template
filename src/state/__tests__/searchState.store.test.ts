import { HierarchyClient as axios } from '../../util/axios';
import { useSearchStore } from '../searchState.store';

describe('search store', () => {
	test('search store is defined', () => {
		expect(useSearchStore.getState()).toMatchSnapshot();
	});

	test('set search loading sets searchLoading', () => {
		// const initialState = useSearchStore.getState();
		useSearchStore.getState().setSearchLoading(true);
		// useSearchStore.setState({ ...initialState, searchLoading: true }, true);

		expect(useSearchStore.getState().searchLoading).toBe(true);
	});

	describe('search Application ', () => {
		test('search application makes http request and sends keyword as query parameter', async () => {
			const store = useSearchStore.getState();

			jest
				.spyOn(axios, 'get')
				.mockImplementationOnce(() => Promise.resolve({ items: [] }));
			store.searchApplication('dummy');
			expect(axios.get).toHaveBeenCalledWith(
				'applications/0/paths/?keyword=dummy',
			);
		});

		test('updates searchStore.searchData with response from server', async () => {
			// const store = useSearchStore.getState();

			jest
				.spyOn(axios, 'get')
				.mockImplementationOnce(() =>
					Promise.resolve({ data: { data: { items: ['dummy'] } } }),
				);

			await Promise.resolve(
				useSearchStore.getState().searchApplication('dummy'),
			);
			// console.log(useSearchStore.getState());

			expect(true).toBe(true);

			// expect(useSearchStore.getState().searchData).resolves.toContain('dummy');
		});

		// 	test('when search is done it updates searchStore.searchLoading to false', () => {});
	});
});
