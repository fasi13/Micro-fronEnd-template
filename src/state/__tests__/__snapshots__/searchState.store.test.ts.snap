// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search store search store is defined 1`] = `
Object {
  "activeNodeId": 0,
  "searchApplication": [Function],
  "searchData": Array [],
  "searchLoading": false,
  "searchedKeyWord": "",
  "setSearchLoading": [Function],
}
`;
